util/fibers/detail/scheduler.cc

```diff
@@ -325,25 +325,34 @@ void Scheduler::ProcessRemoteReady(FiberInterface* active) {
 
     DCHECK(fi->scheduler_ == this);
 
-    // Remote thread can add the same fiber exactly once to the remote_ready_queue.
-    // This is why each fiber is removed from its waitqueue and added to the remote queue
-    // under the same lock.
-    DCHECK(!fi->list_hook.is_linked());
-
-    // ProcessRemoteReady can be called by a FiberInterface::PullMyselfFromRemoteReadyQueue
-    // i.e. it is already active. In that case we should not add it to the ready queue.
+    // Generally, fi should not be in the ready queue if it's still in the remote queue.
+    // Because being in the remote queue means fi is still registered in the wait_queue of
```

Comment:
```suggestion
    // Generally, fi should not be in the ready queue if it's still in the remote queue,
    // because being in the remote queue means fi is still registered in the wait_queue of
```